url8551e9ebc1_mishika05_Salesforce_DataGen_tgz-pJ8-org_json_TestJ8/src/MyProperties.java

```diff
@@ -0,0 +1,51 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+ 
+/**
+ * The Constants class provides properties for SalesforceClient.java,  Record.java, WorkRecord.java and Bug.java
+ * @author  Mishika Vora
+ */
+public class MyProperties {
+    public String environment;
+    public String callbackUrl;
+    public String CLIENT_ID;
+    public String CLIENT_SECRET;
+    public String USERNAME;
+    public String PASSWORD;
+    public int numRecs;
+	public int numThreads;
+    private static Logger logger = Logger.getLogger(MyProperties.class);
+    
+    
+    /**
+     * Creates a new MyProperties object and sets all fields
+     * @param propFile - properties file to refer to
+     */
+    public MyProperties(String propFile) {
+    	Properties prop = new Properties();
+    	try {
+            //load a properties file
+    		prop.load(new FileInputStream(propFile));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).